### PR TITLE
Split node edge logic from dot

### PR DESCRIPTION
**Pull request by Lee-W**
* Split the logic or creating node and edge from `dot.py` so that we can have other presentation beside dot file
* Move the examples in `test/` to `examples/`

### CODE REVIEW

File: awaitwhat/node.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    stops = {task: blockers(task) for task in tasks}
+    nodes = set()
+    edges = list()
+    for task, tasks_to_wait in stops.items():
+        source_node = build_node(task, current)
+        nodes.add(source_node)
+        for task_to_wait in tasks_to_wait:
+            dest_node = build_node(task_to_wait, current)
+            nodes.add(dest_node)
+            edges.append(f'{id(source_node.task)} -> {id(dest_node.task)}')
```
Comment (dimaqq) [inline awaitwhat/node.py]: `123 -> 456` is graphviz format, that part should go to `dot.py`.

Comment (Lee-W) [reply awaitwhat/node.py]: Yes, I just move it to `dot.py`. Thanks!

File: awaitwhat/dot.py [hunk 2]
```diff
@@ -53,23 +15,20 @@ def dumps(tasks):
...
-        current = None
-    stops = {t: blockers(t) or [f"<Not blocked {random.random()}>"] for t in tasks}
-    nodes = set(sum(stops.values(), list(stops.keys())))
-    nodes = "\n        ".join(f"{id(t)} {describe(t, current)}" for t in nodes)
+    prefix = '\n        '
 
-    edges = "\n        ".join(
-        f"{id(k)} -> {', '.join(str(id(v)) for v in vv)}" for k, vv in stops.items()
+    nodes, edges = build_nodes_and_edges(tasks)
+    node_labesls = prefix.join(
```
Comment (freedomofkeima) [inline awaitwhat/dot.py]: Nitpick: s/labesls/labels

File: awaitwhat/blocker.py [hunk 3]
```diff
@@ -1,11 +1,12 @@
 import inspect
+import random
 
 
 def blockers(task):
     """What does this task wait for?"""
     w = task._fut_waiter
     if not w:
-        return
+        return [f'<Not blocked {random.random()}>']
```
Comment (freedomofkeima) [inline awaitwhat/blocker.py]: @dimaqq Should we merge https://github.com/dimaqq/awaitwhat/pull/15 first?

File: awaitwhat/node.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+def build_node(task, current):
+    if isinstance(task, asyncio.Task):
+        buf = io.StringIO()
+        task_print_stack(task, None, buf)
+        try:
+            name = task.get_name()
+        except AttributeError:
+            name = "Task"
+        # FIXME can we ever see a "done" task?
```
Comment (freedomofkeima) [inline awaitwhat/node.py]: Unnecessary comment, fixed in #12

File: awaitwhat/dot.py (hunk 2 above)
Comment (Lee-W) [reply awaitwhat/dot.py]: Fixed. Thanks!

File: awaitwhat/node.py (hunk 4 above)
Comment (Lee-W) [reply awaitwhat/node.py]: I've just fixed that. Thanks!

File: awaitwhat/blocker.py (hunk 3 above)
Comment (dimaqq) [reply awaitwhat/blocker.py]: Maybe it's easier to merge the name change later...

**Pull request merged**